projects/com.oracle.truffle.llvm.parser.impl/src/com/oracle/truffle/llvm/parser/impl/LLVMVisitor.java

```diff
@@ -1006,6 +1021,15 @@ private LLVMExpressionNode visitAliasConstant(Alias ref) {
         if (aliaseeRef instanceof FunctionHeader) {
             LLVMFunctionDescriptor function = createLLVMFunctionFromHeader((FunctionHeader) aliaseeRef);
             return factoryFacade.createLiteral(function, LLVMBaseType.FUNCTION_ADDRESS);
+        } else if (aliaseeRef instanceof GlobalVariable) {
+            GlobalVariable originalVar = (GlobalVariable) aliaseeRef;
+            Object globalVar = globalVars.get(originalVar);
+            LLVMParserAsserts.assertNotNull(globalVar);
+            return factoryFacade.createLiteral(globalVar, LLVMBaseType.ADDRESS);
+        } else if (aliaseeRef instanceof Alias) {
+            Object globalVar = aliases.get(ref);
+            LLVMParserAsserts.assertNotNull(globalVar);
```

Comment:
I am not sure if that would be correct. The representation of an allocation is independent from the parser implementation, and implemented as class implementing the interface `NodeFactoryFacade`. It would not be impossible, that an implementation just returns global objects and aliases and resolves them during runtime.